lemminx-maven/src/main/java/org/eclipse/lemminx/maven/DOMUtils.java

```diff
@@ -61,6 +64,29 @@ public static DOMNode findClosestParentNode(final IPositionRequest request, fina
 		}
 		return foundNodes;
 	}
+	
+	/**
+	 * Gets the range for the specified text parameter within an IPositionRequest's
+	 * node
+	 * 
+	 * @param request
+	 * @param text
+	 * @return the range of the specificed text within the IPositionRequest's node,
+	 *         or null if the text could not be found in the node.
+	 * @throws DOMException
+	 * @throws BadLocationException
+	 */
+	public static Range getTextRange(IPositionRequest request, String text) throws DOMException, BadLocationException {
```

Comment:
I believe this would fail in some case, for example if I use `<blah>${property} ${property}</blah>`, then a request on the 2nd property would result on the range of the 1st one (if I' not mistaken).
Maybe `getMavenPropertyInRequest` should instead return a `Pair<Range, String>` where the range would be the actual range under the request that match a property.

Reply:
@mickaelistria I see the issue your mentioning and agree it should be fixed for this case. However I'm not sure I understand your proposed solution unfortunately. Could you give a more detailed explanation or example?

Reply:
The operation that finds a property name for a given position is actually iterating on the string and this same iteration can be used to find the range. As it appears the range is important, and that just looking for a substring is too weak, then we should also return the range when returning the property name.